bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,135 @@
+
+        $allowedAll = $this->allContentTypes;
+        $allowedContentTypes = $this->allowedContentTypes;
+        $contentType = $content->getContentType();
+        if (!$allowedAll && !in_array($contentType->identifier, $allowedContentTypes, true)) {
+            return false;
+        }
+
+        $fieldDefinitions = $contentType->getFieldDefinitions();
+        if (!$fieldDefinitions->has('publish_from') || !$fieldDefinitions->has('publish_to')) {
+            return false;
+        }
+
+        $publishFromField = $content->getField('publish_from');
+        $publishToField = $content->getField('publish_to');
+        if (
+            ($publishFromField->fieldTypeIdentifier !== 'ezdatetime' && $publishFromField->fieldTypeIdentifier !== 'ezdate')
+            || ($publishToField->fieldTypeIdentifier !== 'ezdatetime' && $publishToField->fieldTypeIdentifier !== 'ezdate')
+        ) {
+            return false;
+        }
+
+        $publishFromDateTime = $this->getDateTime($publishFromField);
+        $publishToDateTime = $this->getDateTime($publishToField);
+        if ($publishFromDateTime !== null && $publishToDateTime !== null && $publishToDateTime <= $publishFromDateTime) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function shouldHide(Content $content): bool
+    {
+        $publishFromField = $content->getField('publish_from');
+        $publishToField = $content->getField('publish_to');
+
+        $publishFromDateTime = $this->getDateTime($publishFromField);
+        $publishToDateTime = $this->getDateTime($publishToField);
+
+        $currentDateTime = new DateTime();
+        if (($publishFromDateTime !== null && $publishFromDateTime > $currentDateTime)
+            || ($publishToDateTime !== null && $publishToDateTime <= $currentDateTime)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function shouldReveal(Content $content): bool
+    {
+        $publishFromField = $content->getField('publish_from');
+        $publishToField = $content->getField('publish_to');
+
+        $publishFromDateTime = $this->getDateTime($publishFromField);
+        $publishToDateTime = $this->getDateTime($publishToField);
+
+        if ($publishToDateTime === null && $publishFromDateTime === null) {
+            return true;
+        }
+
+        $currentDateTime = new DateTime();
+        if (($publishFromDateTime !== null && $publishFromDateTime <= $currentDateTime
+            && ($publishToDateTime === null || $publishToDateTime > $currentDateTime))
+            || ($publishFromDateTime === null && $publishToDateTime > $currentDateTime)
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function getDateTime(Field $field): null|DateTime
+    {
+        if ($field->value instanceof Value) {
+            return $field->value->date;
+        }
+        if ($field->value instanceof \Ibexa\Core\FieldType\DateAndTime\Value) {
```

Comment:
Import the `Value`.

Reply:
fixed in https://github.com/netgen/ibexa-scheduled-visibility/pull/1/commits/f10782221740315719914d41ada84848d384f31c